client/packages/common/src/ui/layout/tables/material-react-table/useMaterialTableColumns.tsx

```diff
@@ -37,41 +27,34 @@ export const useMaterialTableColumns = <T extends MRT_RowData>(
     const filterUpdaters: Record<string, (value: any) => void> = {};
 
     columns.forEach(
-      ({ filterType = 'none', filterValues, ...mrtProperties }) => {
+      ({ filterType = 'none', filterValues, align, ...mrtProperties }) => {
         const filterKey = (mrtProperties.accessorKey ||
           mrtProperties.id) as string;
 
+        const columnDefProperties: Partial<MRT_ColumnDef<T>> = {};
+
+        // Add canonical filter definitions
         switch (filterType) {
           case 'none':
-            mrtColumnDefinitions.push({
-              ...mrtProperties,
-              enableColumnFilter: false,
-            });
+            columnDefProperties.enableColumnFilter = false;
             break;
           case 'text':
-            mrtColumnDefinitions.push(mrtProperties);
             filterUpdaters[filterKey] = (value: string) => {
               updateQuery({ [filterKey]: value });
             };
             break;
 
           case 'enum':
-            mrtColumnDefinitions.push({
-              ...mrtProperties,
-              filterVariant: 'select',
-              filterSelectOptions: filterValues,
-            });
+            columnDefProperties.filterVariant = 'select';
+            columnDefProperties.filterSelectOptions = filterValues;
```

Comment:
I kind of think we could just use filterVariant and filterSelectOptions in the column definitions themselves? They're still clear from the naming what they do, and means they will be searchable fields in the docs, less for us to handle?

If you agree, I can remove in my PR after pulling this in

Reply:
The reason I called them something different is because of the additional logic applied in the columns hook -- when you specify `filterType: enum`, it does a very specific filter mapping (for updating and parsing the URL query) as well as generating the `filterVariant: select` value. I thought this was better, cos `filterVariant: select` could map to multiple filter configurations. If you use the native properties, it's not clear if you're wanting our specific filter mapping, or defining a custom one. Whereas with my way, you still have the *option* to override the resulting filters (using native properties) *as well as* getting a specific filter-type mapping (or bypass our mapping completely if you want).

Reply:
Ah ok thanks for context - I might still have a bit of a play, as I think the distinction is actually in whether or not we have `manualFiltering` enabled for the table? If we do, then we'll want any filterable columns to use the URL query etc, but our filter mapping won't have any effect on MRTs default table filtering.